worker/backend/container_test.go

```diff
@@ -0,0 +1,185 @@
+package backend_test
+
+import (
+	"errors"
+
+	"code.cloudfoundry.org/garden"
+	"github.com/concourse/concourse/worker/backend"
+	"github.com/concourse/concourse/worker/backend/backendfakes"
+	"github.com/concourse/concourse/worker/backend/libcontainerd/libcontainerdfakes"
+	"github.com/containerd/containerd"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ContainerSuite struct {
+	suite.Suite
+	*require.Assertions
+
+	container           *backend.Container
+	containerdContainer *libcontainerdfakes.FakeContainer
+	containerdProcess   *libcontainerdfakes.FakeProcess
+	containerdTask      *libcontainerdfakes.FakeTask
+	rootfsManager       *backendfakes.FakeRootfsManager
+	killer              *backendfakes.FakeKiller
+}
+
+func (s *ContainerSuite) SetupTest() {
+	s.containerdContainer = new(libcontainerdfakes.FakeContainer)
+	s.containerdProcess = new(libcontainerdfakes.FakeProcess)
+	s.containerdTask = new(libcontainerdfakes.FakeTask)
+	s.rootfsManager = new(backendfakes.FakeRootfsManager)
+	s.killer = new(backendfakes.FakeKiller)
+
+	s.container = backend.NewContainer(
+		s.containerdContainer,
+		s.killer,
+		s.rootfsManager,
+	)
+}
+
+// func (s *ContainerSuite) TestStopWithKillUngracefullyStops() {
```

Comment:
Replace these tests with tests for the new Killer

Reply:
https://github.com/concourse/concourse/issues/5184